app/models/journeys/adjustments/WhichYearIsLossReported.scala

```diff
@@ -21,16 +21,24 @@ import play.api.i18n.Messages
 import uk.gov.hmrc.govukfrontend.views.Aliases.Text
 import uk.gov.hmrc.govukfrontend.views.viewmodels.radios.RadioItem
 
+import java.time.Year
+
 sealed trait WhichYearIsLossReported
 
 object WhichYearIsLossReported extends Enumerable.Implicits {
 
-  case object Year2018to2019 extends WithName("2018to2019") with WhichYearIsLossReported
-  case object Year2019to2020 extends WithName("2019to2020") with WhichYearIsLossReported
+  case object currentTaxYearMinusFour  extends WithName(generateYears().last) with WhichYearIsLossReported
```

Comment:
we changed hardcoded 2018to2019 with generateYears() - but it is still hardcoded. The application need to be restarted if we want to support more years. I wouldn't use generateYears as enum.

We have two choices:
- We hardcode like it was but for more years, and each year we remove the oldest one and add newer one. Not perfect, but easy to implement.
- We move away from enum as the value is dynamic always 5 tax years.